src/lex.cpp

```diff
@@ -644,6 +667,19 @@ void lexer::parse_current_token() {
   }
 }
 
+bool lexer::test_for_regexp(int offset) {
+  lexer_transaction transaction = this->begin_transaction();
+
+  this->last_token_.type = token_type::slash;
+  this->input_ += offset + 1;
```

Comment:
Nit: Instead of passing the offset, I think it makes more sense to pass a pointer instead:
```suggestion
bool lexer::test_for_regexp(const char8* regexp_begin) {
  lexer_transaction transaction = this->begin_transaction();

  this->last_token_.type = token_type::slash;
  this->input_ = regexp_begin;
```